### PR TITLE
Grinder - Use items for full container, not all

### PR DESCRIPTION
**Pull request by botka4aet**
## What Does This PR Do
- Now can grind rods
- Don't use all grinding stack items. EX: you place 50 stack plasma. Grind only 5 of them for 100 Plasma dust
*changed code from //Sheets and rods(!)

## Why It's Good For The Game
- Grinder machine works as it shoud be

## Changelog
:cl:
tweak: All-In-One Grinder can grind rods
tweak: All-In-One Grinder don't grind the entire stack and instead grind items until it is full.
/:cl:

### CODE REVIEW

**Comment by FlattestGuitar** [comment]
If you're changing the indentation of a whole file you should put it in a separate commit. This is needlessly hard to review.

**Comment by variableundefined** [comment]
A more informative PR title would be appreciated

**Review by S34NW — CHANGES_REQUESTED** [review]
This isn't really on you because of how the surrounding file is a big mess of spacing, but we generally don't include spaces after `if` and `for` checks. Best to tidy up the ones you touch here.

**Comment by Aurorablade** [comment]
S34NW got that at least but the indenetation thing is fine after i weNT THROUGH it and everything it looks okay but does bare someone testing it just to make sure all wheels running at proper amount of squeakyness.

File: code/modules/reagents/chemistry/machinery/reagentgrinder.dm [hunk 1]
```diff
@@ -303,213 +304,218 @@
...
+	for (var/obj/item/stack/O in holdingitems)
+		if (beaker.reagents.holder_full())
+			break
+
+		var/allowed = get_allowed_by_id(O)
+		if(isnull(allowed))
+			break
+
+		var/space = beaker.reagents.maximum_volume - beaker.reagents.total_volume
+		while(O.amount) //Work untill hold something
```
[the remark this answers is not in the document]
Comment (variableundefined) [reply code/modules/reagents/chemistry/machinery/reagentgrinder.dm:472]: Change the comment to "Grind until there's no more reagents"

**Review by variableundefined — CHANGES_REQUESTED** [review]
Awesome PR, the grinder has always been very unintuitive and wasteful. This will improve QOL for chemistry tonnes. 

This is good to go once you fix up the styling issues. If you are on Discord, ping me within the weekend.

**Comment by variableundefined** [comment]
I'll take the liberty to change the changelog and PR title a bit if that is fine with you @botka4aet

**Comment by botka4aet** [comment]
> I'll take the liberty to change the changelog and PR title a bit if that is fine with you @botka4aet

Sure. All for comfy game

**Comment by botka4aet** [comment]
Syntax fix and move remove_object to cicle end

**Review by variableundefined — APPROVED** [review]

**Pull request merged**